### PR TITLE
validating refernceId backend

### LINKED ISSUE #631: Add family form - validate refernce ID - Backend
Backend validating of referenceId

### PR DESCRIPTION
**Pull request by En-Lun**
Validating referenceId backend.

added check in controller and a function in the repository

close #631

### CODE REVIEW

**Review by hildmi — APPROVED** [review]

File: GiEnJul/Repositories/RecipientRepository.cs [hunk 1]
```diff
@@ -44,6 +45,16 @@ public async Task<List<Models.Recipient>> GetUnmatchedRecipientsAsync(string loc
             return _mapper.Map<List<Models.Recipient>>(recipients);
         }
 
+        public async Task<bool> RecipientDoesExist(string referenceId)
+        {
+            var query = $"ReferenceId eq '{referenceId}";
+            var exists = await GetAllByQueryAsync(query);
+            if (_mapper.Map<List<Models.Recipient>>(exists).Count > 0) return true;
```
Comment (RonHaland) [inline GiEnJul/Repositories/RecipientRepository.cs:52]:
```suggestion
            if (_mapper.Map<List<Models.Recipient>>(exists).Any()) 
                 return true;
```
But I think this change also requires "using linq" 🤷‍♂️
Anyways, please move the return down

File: GiEnJul/Repositories/RecipientRepository.cs [hunk 2]
```diff
@@ -44,6 +45,16 @@ public async Task<List<Models.Recipient>> GetUnmatchedRecipientsAsync(string loc
...
+           
+            else return false;
```
Comment (RonHaland) [inline GiEnJul/Repositories/RecipientRepository.cs:54]:
```suggestion
            else 
                 return false;
```
Move these down

**Review by RonHaland — COMMENTED** [review]
Just some visual feedback 😁

**Review by MagnusVG — DISMISSED** [review]

File: GiEnJul/Repositories/RecipientRepository.cs (hunk 1 above)
Comment (En-Lun) [reply GiEnJul/Repositories/RecipientRepository.cs]: fixed

File: GiEnJul/Repositories/RecipientRepository.cs (hunk 2 above)
Comment (En-Lun) [reply GiEnJul/Repositories/RecipientRepository.cs]: fixed

**Review by MagnusVG — APPROVED** [review]

**Pull request merged**